### PR TITLE
fix: hack to output OCI v1.1 artifact

### PR DESCRIPTION
**Pull request by tri-adam**
## Description of the Pull Request (PR):

Data containers in Singularity use an empty JSON document as their "config". Thus, based on the [OCI v1.1 release notes](https://opencontainers.org/posts/blog/2024-03-13-image-and-distribution-1-1/), they're a good candidate to make use of the `artifactType` field.

The `google/go-containerregistry` does not currently include `artifactType` in `v1.Manifest`, as per the OCI v1.1 specification. There is an open PR to address that (https://github.com/google/go-containerregistry/pull/1931) but it hasn't had any movement in a while. This PR works around the lack of support until that field is added in the `google/go-containerregistry` module.

### This fixes or addresses the following GitHub issues:

 - Fixes #

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/main/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/main/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/main/CONTRIBUTING.md)
- Added tests to validate this PR, linted with `make check`  and tested this PR locally with a `make test`, and `make testall` if possible (see CONTRIBUTING.md).
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/main/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/main/CONTRIBUTORS.md)

### CODE REVIEW

File: internal/pkg/ocisif/datacontainer.go [hunk 1]
```diff
@@ -158,3 +167,68 @@ func DataContainerLayerOffset(f *os.File) (int64, error) {
...
+
+	var manifest struct {
+		SchemaVersion int64             `json:"schemaVersion"`
+		MediaType     types.MediaType   `json:"mediaType,omitempty"`
+		ArtifactType  string            `json:"artifactType,omitempty"`
+		Config        v1.Descriptor     `json:"config"`
+		Layers        []v1.Descriptor   `json:"layers"`
+		Annotations   map[string]string `json:"annotations,omitempty"`
+		Subject       *v1.Descriptor    `json:"subject,omitempty"`
+	}
```
Comment (wobito) [inline internal/pkg/ocisif/datacontainer.go:215]: Can't simplify this a bit with?

```suggestion
	var manifest struct {
		v1.Manifest
		ArtifactType  string            `json:"artifactType,omitempty"`
	}
```

**Review by wobito — COMMENTED** [review]
just noticed one thing

File: internal/pkg/ocisif/datacontainer.go (hunk 1 above)
Comment (tri-adam) [reply internal/pkg/ocisif/datacontainer.go:215]: That's definitely simpler, but I think that will change the order of the fields? That doesn't matter at the JSON level, but that would mean that we'd generate a different manifest digest than GGCR likely will (if/when https://github.com/google/go-containerregistry/pull/1931 lands.)

Comment (wobito) [reply internal/pkg/ocisif/datacontainer.go:215]: That makes sense

**Review by dtrudg — APPROVED** [review]
LGTM. Many thanks for this!

**Comment by tri-adam** [comment]
Just had one more thought this morning, we are setting the `artifactType` to `application/vnd.sylabs.data-container.config.v1+json`... but the artifact itself is not a `config`, nor is it JSON document? The layer media types describe the format of the layer, so I don't think we need to describe that here, but perhaps this should just be something like `application/vnd.sylabs.data-container.v1`?

**Comment by dtrudg** [comment]
> Just had one more thought this morning, we are setting the `artifactType` to `application/vnd.sylabs.data-container.config.v1+json`... but the artifact itself is not a `config`, nor is it JSON document? The layer media types describe the format of the layer, so I don't think we need to describe that here, but perhaps this should just be something like `application/vnd.sylabs.data-container.v1`?

Ahh, yeah. That would make sense without the `config` 👍

**Pull request merged**